armi/bookkeeping/historyTracker.py

```diff
@@ -331,8 +332,7 @@ def getTimeIndices(self, a=None, boc=False, moc=False, eoc=False):
                     )
                 # only add this timestep if it's around for this assembly.
                 if blockLocationLabel is not None:
-                    bLoc = locations.Location(label=blockLocationLabel)
-                    if bLoc.isInCore():
+                    if a.spatialLocator.grid is coreGrid:
```

Comment:
This condition can be checked at the top of the method right? rather than way down here?

Reply:
I... don't know. I actually think this is broken in general, but only updated it so that it will be less confusing to the next person to come along. If you look at `_blockLocationAtTimenode()`, its making pretty bold assumptions about the database structure:
```
    def _blockLocationAtTimenode(self, block, timeNode):
        """Find block location label at a specific timenode."""
        dbi = self.getInterface("database")
        ids = dbi.database.readBlockParam("id", timeNode)
        locs = dbi.database.lookupGeometry()
```.

With the database-based history tracking, a lot of this can maybe go anyways, but that should be future work; I will add some comments to this effect.